.github/workflows/integration.yaml

```diff
@@ -0,0 +1,157 @@
+name: Integration testing
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/integration.yaml'
+      - 'gothemis/**'
+      - 'src/soter/**'
+      - 'src/themis/**'
+      - 'src/wrappers/themis/jsthemis/**'
+      - 'src/wrappers/themis/php/**'
+      - 'src/wrappers/themis/php7/**'
+      - 'src/wrappers/themis/python/**'
+      - 'src/wrappers/themis/ruby/**'
+      - 'src/wrappers/themis/rust/**'
+      - 'src/wrappers/themis/wasm/**'
+      - 'third_party/boringssl/src/**'
+      - 'tools/_integration/**'
+      - 'Cargo.toml'
+      - '**/*.mk'
+      - 'Makefile'
+      - '!**/README*'
+  push:
+    branches:
+      - master
+      - stable
+      - release/*
+  schedule:
+    - cron: '0 6 * * *' # every day at 6:00 UTC
+
+env:
+  WITH_FATAL_WARNINGS: yes
+
+jobs:
+  cross-language:
+    name: Cross-language tests
+    runs-on: ubuntu-latest
+    env:
+      GOTHEMIS_IMPORT: github.com/cossacklabs/themis/gothemis
+    steps:
+      - name: Install system dependencies
+        run: |
+          sudo sh -c 'echo "DEBIAN_FRONTEND=noninteractive" >> /etc/environment'
+          sudo apt update
+          sudo apt install --yes gcc g++ make libssl-dev \
+            python  python-setuptools \
+            python3 python3-setuptools \
+            ruby ruby-dev \
+            pkg-config clang
+      - name: Install RVM
+        run: |
+          sudo apt install --yes software-properties-common
+          sudo apt-add-repository -y ppa:rael-gc/rvm
+          sudo apt update
+          sudo apt install rvm
+          # Recent versions of RVM do not add us to "rvm" group automatically
+          # and install their binaries into /usr/share, expecting the PATH
+          # to be set via shell profile. GitHub Actions do not load profile
+          # so we have to tweak the path manually here.
+          sudo usermod -a -G rvm $(id -nu)
+          echo "::add-path::/usr/share/rvm/bin"
```

Comment:
for what this command?

Reply:
This adds `rvm` command into PATH for all future actions. For some weird reason RVM is really unfriendly to automated installation. I’ve copied this from CircleCI scripts, but I wonder if it can be simplified. I guess we could avoid using RVM entirely and install Ruby via some GitHub Action way. But in this PR I’d avoid tweaking it.

Reply:
actually I dont understand how `echo "::add-path::/path/"` adds path to $PATH. I know `export PATH=/new/path:$PATH` way. And didn't found anything in google how to add a new path with `echo "..."` command without append to some profile file
p.s. as I remember, was some bug or we need some feature in ruby which was fixed/added in new minor version of ruby but debian hasn't it and used some own patches. and the simplest method to install supported ruby version was rvm.

Reply:
> actually I dont understand how `echo "::add-path::/path/"` adds path to $PATH

Oh... I'm sorry. This is some GitHub Actions magic, [here are the docs](https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#adding-a-system-path). It's one of the special “workflow commands”. Each action executes in a separate shell session so `export PATH` works only for next command in an action but not for the following actions. `echo ::add-path` adds the path for next actions (but not for the current one).

> as I remember, was some bug or we need some feature in ruby which was fixed/added in new minor version of ruby but debian hasn't it and used some own patches

Currently even CircleCI does not test any particular Ruby, we're testing against the system one only. I'd imagine that BuildBot uses it though.

I've looked through our internal task tracker but that did not reveal anything relevant.

Whatever. I'll look into properly installed Ruby later. We'd better test all supported versions, which [currently are](https://www.ruby-lang.org/en/downloads/branches/) 2.4 — 2.7.

Reply:
> GitHub Actions magic

holy shit, one more tool specific language ))

Reply:
How else are you expected to differentiate yourself from other free\* CI services? Obviously not by providing better computing power or more stable platform, but by locking your users in with yet another YAML syntax.

For that reason I don’t *really* like all those special actions to install things. However, they do tend to integrate better.